presto-hive/src/test/java/com/facebook/presto/hive/AbstractTestOptimizedPartitionedOutputOperatorQueries.java

```diff
@@ -0,0 +1,560 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.tests.AbstractTestDistributedQueries;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.SystemSessionProperties.OPTIMIZED_REPARTITIONING_ENABLED;
+import static org.testng.Assert.assertEquals;
+
+public abstract class AbstractTestOptimizedPartitionedOutputOperatorQueries
+        extends AbstractTestDistributedQueries
+{
+    protected AbstractTestOptimizedPartitionedOutputOperatorQueries(QueryRunnerSupplier supplier)
+    {
+        super(supplier);
+    }
+
+    @Override
+    public void testDelete()
+    {
+        // This connector only supports delete where one or more partitions are deleted entirely
+    }
+
+    @Override
+    public void testInsertIntoNotNullColumn()
+    {
+        // Catalog 'hive' does not support non-null column for column name 'column_b'
+    }
+
+    @Test
+    public void testBoolean()
```

Comment:
These new tests don't seem to be specific to partitioned output operator. Consider moving them into `AbstractTestDistributedQueries`.